src/core/ddsi/src/ddsi_proxy_participant.c

```diff
@@ -682,6 +716,7 @@ static void delete_ppt (struct ddsi_proxy_participant *proxypp, ddsrt_wctime_t t
   }
   ddsrt_free (child_entities);
 
+  maybe_update_as_disc_for_proxypp (proxypp->e.gv, proxypp->as_meta, isimplicit ? MUADFPOP_REMOVE_ON_EXPIRY : MUADFPOP_REMOVE_ON_DELETE);
```

Comment:
I think your confusion is caused by `implicit` being a bad choice of word here. It is actually a flag indicating lease expiry ... and it makes sense (to me anyway) that a lease expiry is potentially handled differently from removing something you know no longer exists.

In this particular case the difference is: if an address used for this participant is only known because a participant was discovered there at some point, then (1) in the case of lease expiry we should keep pinging that address for some time in case it was a temporary disconnect because we don't have evidence that the other will keep pinging us; and (2) in the case of the participant being deleted, we can drop the address immediately on the assumption that a new participant at that address will start pinging us so we can rediscover the address.

Yes, there are assumptions here on how things behave and are configured ... but we can't just keep pinging every address we discover forever.